src/battle_util.c

```diff
@@ -388,13 +377,24 @@ static bool32 IsUnnerveAbilityOnOpposingSide(u32 battler)
     return FALSE;
 }
 
+static inline bool32 IsDragonDartsSecondHit(void)
+{
+    if (GetBattlerMoveTargetType(gBattlerAttacker, gCurrentMove) != TARGET_SMART)
+        return FALSE;
+
+    if (gMultiHitCounter == 1)
```

Comment:
```suggestion
    if (gMultiHitCounter < GetMoveStrikeCount(gCurrentMove))
```
Since `TARGET_SMART` requires `strikeCount > 1`, this accomodates for `strikeCount > 2`.

Also should `battlerAtk` and `move` be passed to this function instead of checking `gBattlerAttacker`/`gCurrentMove`?